flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/main/java/com/ververica/cdc/connectors/starrocks/sink/StarRocksUtils.java

```diff
@@ -0,0 +1,333 @@
+        }
+        tableBuilder.setTableProperties(tableConfig.getProperties());
+        return tableBuilder.build();
+    }
+
+    /** Convert CDC data type to StarRocks data type. */
+    public static void toStarRocksDataType(Column cdcColumn, StarRocksColumn.Builder builder) {
+        CdcDataTypeTransformer dataTypeTransformer = new CdcDataTypeTransformer(builder);
+        cdcColumn.getType().accept(dataTypeTransformer);
+    }
+
+    /** Format DATE type data. */
+    private static final SimpleDateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd");
+
+    /** Format timestamp-related type data. */
+    private static final DateTimeFormatter DATETIME_FORMATTER =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    /**
+     * Creates an accessor for getting elements in an internal RecordData structure at the given
+     * position.
+     *
+     * @param fieldType the element type of the RecordData
+     * @param fieldPos the element position of the RecordData
+     */
+    public static RecordData.FieldGetter createFieldGetter(DataType fieldType, int fieldPos) {
+        final RecordData.FieldGetter fieldGetter;
+        // ordered by type root definition
+        switch (fieldType.getTypeRoot()) {
+            case BOOLEAN:
+                fieldGetter = record -> record.getBoolean(fieldPos);
+                break;
+            case TINYINT:
+                fieldGetter = record -> record.getByte(fieldPos);
+                break;
+            case SMALLINT:
+                fieldGetter = record -> record.getShort(fieldPos);
+                break;
+            case INTEGER:
+                fieldGetter = record -> record.getInt(fieldPos);
+                break;
+            case BIGINT:
+                fieldGetter = record -> record.getLong(fieldPos);
+                break;
+            case FLOAT:
+                fieldGetter = record -> record.getFloat(fieldPos);
+                break;
+            case DOUBLE:
+                fieldGetter = record -> record.getDouble(fieldPos);
+                break;
+            case DECIMAL:
+                final int decimalPrecision = getPrecision(fieldType);
+                final int decimalScale = getScale(fieldType);
+                fieldGetter =
+                        record ->
+                                record.getDecimal(fieldPos, decimalPrecision, decimalScale)
+                                        .toBigDecimal();
+                break;
+            case CHAR:
+            case VARCHAR:
+                fieldGetter = record -> record.getString(fieldPos).toString();
+                break;
+            case DATE:
+                fieldGetter =
+                        record ->
+                                DATE_FORMATTER.format(
+                                        Date.valueOf(
+                                                LocalDate.ofEpochDay(record.getInt(fieldPos))));
```

Comment:
Is this because of different ways of handling time that this method was rewritten instead of using [createFieldGetter in RecordData](https://github.com/ververica/flink-cdc-connectors/blob/ab3f7d81356579e062d56586700de84b3924cf37/flink-cdc-common/src/main/java/com/ververica/cdc/common/data/RecordData.java#L172)？

Reply:
Yes, we need to convert the data to a string so that it can be sent to starrocks in json format